core/service/src/chain_ops.rs

```diff
@@ -198,6 +205,13 @@ pub fn import_blocks<F, E, R>(
 			queue.poll_actions(cx, &mut link);
 			std::task::Poll::Pending::<Result<(), ()>>
 		}).compat().poll();
+		if link.has_error {
+			info!(
+				"Stopping after #{} blocks because of an error",
+				link.imported_blocks,
+			);
+			return Ok(Async::Ready(()));
```

Comment:
Should it return `Err` instead?

Reply:
That's debatable, depending on what we want the semantics of the function to be.
Considering that the function handles printing out what happens, I went with returning `Ok` as a way to say "we're done with this operation". Once we switch to stable futures you'd just return a `()` instead of a `Result`.

Reply:
To expand, if instead we were to return an `Err`, I would make it so that the `Ok` or the `Err` contains a report of what happens, and it's the caller that would print that out.

Reply:
This is a future that will be spawned and run to completion in one place, so I'm fine with keeping it as it is. It's not composable anyway as is because it prints all the stuff.